lte/gateway/python/magma/subscriberdb/store/sqlite.py

```diff
@@ -35,93 +38,131 @@ class SqliteStore(BaseStore):
     """
 
     def __init__(self, db_location, loop=None):
-        self._db_location = db_location
-        self._tlocal = threading.local()
+        self._db_location = self._create_db_locations(db_location)
         self._create_store()
         self._on_ready = OnDataReady(loop=loop)
 
-    @property
-    def conn(self):
-        """
-        Returns a thread local connection to the sqlite db.
-        """
-        if not getattr(self._tlocal, 'conn', None):
-            self._tlocal.conn = sqlite3.connect(self._db_location, uri=True)
-        return self._tlocal.conn
+
+    def _create_db_locations(self, db_location):
+        # db_location in subscriberdb.yml: file:/var/opt/magma/subscriber.db?cache=shared
+        i = db_location.find(".db")
+        db_location_list = []
+        # no sharding if it is not a file
+        if i == -1:
+            return [db_location]
+
+        # file name is passed, use it as a base
+        for n_shard in range(N_SHARDS):
+            if n_shard < 10: # prefix i with '0'
+                db_location_list.append(db_location[:i] + '0' + str(n_shard) + db_location[i:])
+            else:
+                db_location_list.append(db_location[:i] + str(n_shard) + db_location[i:])
```

Comment:
Maybe you can use "{:02d}".format(n_shard)

Reply:
actually I will eliminate this, it is a residual from earlier commits where I wasn't keeping a list...